go/tools/gazelle/packages/package.go

```diff
@@ -15,7 +15,46 @@ limitations under the License.
 
 package packages
 
-import "strings"
+import (
+	"fmt"
+	"strings"
+)
+
+// PlatformConstraints is a map from config_setting labels (for example,
+// "@io_bazel_rules_go//go/platform:linux_amd64") to a sets of build tags
+// that are true on each platform (for example, "linux,amd64").
+type PlatformConstraints map[string]map[string]bool
+
+// ParsePlatformConstraints parses a string that contains all platforms and
+// tags on each platform. The string must contain an even number of fields,
+// separated by whitespace. Even fields (counting from 0) are platform labels.
```

Comment:
Flags that are required to contain whitespace are the spawn of the devil....
Not making the label tag separator distinct from the pair separator makes the parsing harder, makes reading it harder, and requires things like "must be an even number".
Also I think they are inverted in logic from the way most people will want to thing about them, they will want to say "these tags should be this label"
```linux,amd64=@io_bazel_rules_go//go/platform:linux_amd64+darwin,amd64=@io_bazel_rules_go//go/platform:darwin_amd64+windows,amd64=@io_bazel_rules_go//go/platform:windows_amd64```
where the + operator is open to anything that is not shell or label specific I think would be better

Given the regularity of the rules, and the fact that we hope this is a temporary solution, it might be even better to support a pattern instead, or just specify the package where the constraints live and always generate label names using strings.Join(tags, "_")